src/CommonITILObject.php

```diff
@@ -1605,6 +1605,12 @@ public function prepareInputForUpdate($input)
             $input['itilcategories_id_code'] = ITILCategory::getById($cat_id)->fields['code'];
         }
 
+        $location_id = $input['locations_id'] ?? 0;
+        $location_id = (int)$location_id;
+        if ($location_id) {
+            $input['_locations_code'] = Location::getById($location_id)->fields['code'];
```

Comment:
It is safer to check `Location::getById()` result, in case input is not valid.

```suggestion
        if ($location_id > 0 && ($location = Location::getById($location_id)) !== false) {
            $input['_locations_code'] = $location->fields['code'];
```